src/Microsoft.AspNet.Http.Abstractions/Authentication/AuthenticationManager.cs

```diff
@@ -11,9 +12,24 @@ public abstract class AuthenticationManager
     {
         public abstract IEnumerable<AuthenticationDescription> GetAuthenticationSchemes();
 
-        public abstract AuthenticationResult Authenticate(string authenticationScheme);
+        public abstract void Authenticate(AuthenticateContext context);
 
-        public abstract Task<AuthenticationResult> AuthenticateAsync(string authenticationScheme);
+        public abstract Task AuthenticateAsync(AuthenticateContext context);
```

Comment:
I agree, but it's not really intuitive. And you now have to ensure `Accepted` is true:

```csharp
var context = new AuthenticateContext(scheme);
httpContext.Authentication.Authenticate(context);
if (context.Accepted) {
    var principal = context.Principal;
    ...
}
```

... which is not really simpler than the old API:

```csharp
var principal = (await context.Authentication.AuthenticateAsync(scheme))?.Principal;
if (principal != null) {
    ...
}
```